assets/js/components/GoogleChart/utils.js

```diff
@@ -0,0 +1,197 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import set from 'lodash/set';
+import cloneDeep from 'lodash/cloneDeep';
+import merge from 'lodash/merge';
+
+/**
+ * Internal dependencies
+ */
+import { stringToDate } from '../../util/date-range/string-to-date';
+
+/**
+ * Returns the modified data for Google chart.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Array|null}      data          Chart data.
+ * @param {Array|undefined} selectedStats If only certain columns should be displayed for the data set.
+ * @return {Object} The modified data to use in a chart.
+ */
+export const getGoogleChartData = ( data, selectedStats ) => {
+	// Ensure we don't filter out columns that aren't data, but are things like
+	// tooltips or other content.
+	let nonDataColumns = [];
+	if ( data?.length ) {
+		nonDataColumns = data[ 0 ].reduce( ( acc, row, rowIndex ) => {
+			return row?.role ? [ ...acc, rowIndex ] : acc;
+		}, [] );
+	}
+
+	let modifiedData = data;
+	if ( selectedStats?.length > 0 ) {
+		modifiedData = data.map( ( row ) => {
+			return row.filter( ( _columnValue, columnIndex ) => {
+				return (
+					columnIndex === 0 ||
+					selectedStats.includes( columnIndex - 1 ) ||
+					nonDataColumns.includes( columnIndex - 1 )
+				);
+			} );
+		} );
+	}
+
+	return modifiedData;
+};
+
+/**
+ * Returns the optimal height and width for a preview element.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string|undefined} loadingHeight Preview height.
+ * @param {string|undefined} height        Chart height.
+ * @param {string|undefined} loadingWidth  Preview width.
+ * @param {string|undefined} width         Chart width.
+ * @return {Object} The optimal height and width to use in a preview element.
+ */
+export const getLoadingDimentions = (
+	loadingHeight,
+	height,
+	loadingWidth,
+	width
+) => {
+	const dimentions = {
+		height: loadingHeight || height,
+		width: loadingWidth || width,
+	};
+	// If a loading height is set but a width is not (or a loading width is set
+	// but not a height), change the "unset" value to 100% to avoid visual bugs.
+	// See: https://github.com/google/site-kit-wp/pull/2916#discussion_r623866269
+	if ( dimentions.height && ! dimentions.width ) {
+		dimentions.width = '100%';
+	}
+	if ( dimentions.height && ! dimentions.width ) {
+		dimentions.height = '100%';
+	}
+
+	return dimentions;
+};
+
+/**
+ * Returns a combined chart events object.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Array.<Object>|null} chartEvents Event names and its callbacks.
```

Comment:
Based on `GoogleCharts` prop types, it looks like this should be `Array.<Object>|undefined`:

```suggestion
 * @param {Array.<Object>|undefined} chartEvents Event names and its callbacks.
```